tsl/test/sql/merge_chunks.sql

```diff
@@ -0,0 +1,215 @@
+rollback;
+\set ON_ERROR_STOP 1
+
+alter table mergeme set (timescaledb.compress_orderby='time', timescaledb.compress_segmentby='device');
+select compress_chunk('_timescaledb_internal._hyper_1_1_chunk');
+select compress_chunk('_timescaledb_internal._hyper_1_3_chunk');
+
+\set ON_ERROR_STOP 0
+-- cannot merge a compressed chunk
+call merge_chunks('_timescaledb_internal._hyper_1_1_chunk', '_timescaledb_internal._hyper_1_2_chunk');
+call merge_chunks('_timescaledb_internal._hyper_1_2_chunk', '_timescaledb_internal._hyper_1_3_chunk');
+\set ON_ERROR_STOP 1
+
+select decompress_chunk('_timescaledb_internal._hyper_1_3_chunk');
+
+-- Merging with Hypercore TAM works
+select compress_chunk('_timescaledb_internal._hyper_1_1_chunk', hypercore_use_access_method=>true);
+select relname, amname from pg_class cl
+join pg_am am on (cl.relam = am.oid)
+where cl.oid = '_timescaledb_internal._hyper_1_1_chunk'::regclass;
+
+select * from _timescaledb_internal._hyper_1_1_chunk;
+select * from _timescaledb_internal._hyper_1_2_chunk;
+
+begin;
+select * from mergeme;
+call merge_chunks(ARRAY['_timescaledb_internal._hyper_1_1_chunk', '_timescaledb_internal._hyper_1_3_chunk', '_timescaledb_internal._hyper_1_2_chunk']);
+select * from partitions;
+-- Note that no space partition CHECK constraint is added because it
+-- now covers the entire range from -inf to +inf.
+select "Constraint", "Columns", "Expr" from test.show_constraints('_timescaledb_internal._hyper_1_1_chunk');
+select count(*) as num_orphaned_slices from orphaned_slices;
+select * from show_chunks('mergeme');
+
+select relname, amname from pg_class cl
+join pg_am am on (cl.relam = am.oid)
+where cl.oid = '_timescaledb_internal._hyper_1_1_chunk'::regclass;
+select _timescaledb_debug.is_compressed_tid(ctid), * from _timescaledb_internal._hyper_1_1_chunk;
+select * from mergeme;
+rollback;
+
+---
+-- Test some error cases when merging chunks with non-chunks or chunks
+-- from other hypertables
+---
+create table mergeme_too(time timestamptz not null, device int, temp float);
+select create_hypertable('mergeme_too', 'time', 'device', 3, chunk_time_interval => interval '1 day');
+create table mergeme_regular(time timestamptz not null, device int, temp float);
+
+
+insert into mergeme_too values ('2024-01-01', 1, 1.0);
+insert into mergeme_regular select * from mergeme_too;
+
+create materialized view mergeme_mat as
+select * from mergeme_too where device=1;
+
+select * from show_chunks('mergeme_too');
+
+\set ON_ERROR_STOP 0
+-- Merge chunk and regular table
+call merge_chunks('_timescaledb_internal._hyper_1_1_chunk', 'mergeme_regular');
+call merge_chunks('mergeme_regular', '_timescaledb_internal._hyper_1_1_chunk');
+call merge_chunks('_timescaledb_internal._hyper_1_1_chunk', 'mergeme_mat');
+-- Merge chunks from different hypertables
+call merge_chunks('_timescaledb_internal._hyper_1_1_chunk', '_timescaledb_internal._hyper_3_8_chunk');
+\set ON_ERROR_STOP 1
```

Comment:
Two other error cases that might be good to have (to check that they generate a reasonable error, not that they work):

- Merge a table with a "strange" (non-heap and non-hypercore) access method with a chunk of a hypertable. I think it should say that you cannot merge a regular table with a chunk.
- Try to merge two non-adjacent chunks where one has a "strange" table access method